File: sockets/java/src/ClienteBasico.java
import java.io.*;
import java.net.*;
import java.util.Scanner;

/**
 * Cliente básico que se conecta al servidor y envía mensajes
 */
public class ClienteBasico {
    private static final String HOST = "localhost";
    private static final int PUERTO = 8080;
    
    public static void main(String[] args) {
        System.out.println("=== CLIENTE BÁSICO ===");
        System.out.println("Conectando a " + HOST + ":" + PUERTO);
        
        try (
            Socket socket = new Socket(HOST, PUERTO);
            BufferedReader entrada = new BufferedReader(
                new InputStreamReader(socket.getInputStream()));
            PrintWriter salida = new PrintWriter(
                socket.getOutputStream(), true);
            Scanner scanner = new Scanner(System.in)
        ) {
            System.out.println("Conectado al servidor!");
            
            // Leer mensaje de bienvenida del servidor
            String bienvenida = entrada.readLine();
            System.out.println("Servidor: " + bienvenida);
            
            String mensaje;
            while (true) {
                System.out.print("Tu mensaje (o 'salir'): ");
                mensaje = scanner.nextLine();
                
                // Enviar mensaje al servidor
                salida.println(mensaje);
                
                if ("salir".equalsIgnoreCase(mensaje.trim())) {
                    break;
                }
                
                // Leer respuesta del servidor
                String respuesta = entrada.readLine();
                if (respuesta != null) {
                    System.out.println("Servidor: " + respuesta);
                } else {
                    System.out.println("El servidor cerró la conexión");
                    break;
                }
            }
            
        } catch (ConnectException e) {
            System.err.println("No se pudo conectar al servidor. ¿Está ejecutándose?");
            System.err.println("Ejecuta primero: java ServidorBasico");
        } catch (IOException e) {
            System.err.println("Error de conexión: " + e.getMessage());
        }
        
        System.out.println("Cliente desconectado");
    }
}

File: sockets/python/cliente_basico.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Cliente TCP básico que se conecta al servidor
Ejemplo simple de cliente que envía y recibe mensajes
"""

import socket

HOST = 'localhost'
PUERTO = 8080

def main():
    print("=== CLIENTE BÁSICO ===")
    print(f"Conectando al servidor {HOST}:{PUERTO}")
    
    try:
        # Crear socket del cliente y conectar
        cliente = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        cliente.connect((HOST, PUERTO))
        
        print("Conectado al servidor!")
        
        # Recibir mensaje de bienvenida
        bienvenida = cliente.recv(1024).decode('utf-8')
        print(f"Servidor: {bienvenida}", end='')
        
        # Bucle de comunicación
        while True:
            # Leer mensaje del usuario
            mensaje = input("Tú: ")
            
            if not mensaje:
                break
                
            # Enviar mensaje al servidor
            cliente.send(mensaje.encode('utf-8'))
            
            # Recibir respuesta del servidor
            respuesta = cliente.recv(1024).decode('utf-8')
            print(f"Servidor: {respuesta}", end='')
            
            # Si enviamos "adios", salir del bucle
            if mensaje.lower() == "adios":
                break
                
    except ConnectionRefusedError:
        print("No se pudo conectar al servidor.")
        print("Ejecuta primero: python servidor_basico.py")
    except Exception as e:
        print(f"Error de conexión: {e}")
    finally:
        cliente.close()
        print("Desconectado del servidor")

if __name__ == "__main__":
    main()